src/app/client.ts

```diff
@@ -29,11 +29,12 @@ export default class Client extends Base {
     // exchangeType = 'direct';
     // const serializer = 'json';
 
+    const _this = this;
```

Comment:
Can you explain why `_this` is needed?
I don't get the reason and I think it's enough to use `this` below.

Reply:
Unfortunatelly typescript dont correctly handle this object in the embedded function. To correctly access the scope of the main function to the the current object with the CELERY_QUEUE variable, its improtant to store this in a separate variable.
If you dont do this, than you are accessing with "this" the scope of the embedded function.

Reply:
Oh, I got it. Yeah, it would be the issue of `this` scope if we use the embedded function.
But isn't it okay when we use the arrow function in typescript?
And I'm concerned it could be confusing to store `this` in a separate variable.

Reply:
I am not sure how to fix it elegant in TS, but in JS native I would solve it like I did in TS

Reply:
I mean if we use `this`, not storing separate value `_this`, it still does not generate any issue on this code ... does it? 
So could you just rollback `_this` to `this` and add test cases for worker receiving unknown tasks?

Reply:
it will generate an issue.

Reply:
https://www.jackfranklin.co.uk/blog/javascript-variable-scope-this/#:~:text=Today%20I%20want%20to%20talk,they%20exist%20%26%20are%20executed%20in.&text=function%20someFunc()%20%7B%20var%20_this,%22%2C%20function()%20%7B%20console.

Reply:
The link you attached just describes the scope of general function expressions. 
I agree that we should be careful about `this` scope when we use function expression, and it's the reason why we have to use arrow function expressions. 
As I know, the code is using arrow function expressions, so the issues you're concerned about do not occur.  

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Description

Reply:
I know this theory, but check out what your javascript compile create for you instead:

```
Client.prototype.sendTaskMessage = function (taskName, message) {
        var _this = this;
        var headers = message.headers, properties = message.properties, body = message.body, sentEvent = message.sentEvent;
        var exchange = "";
        // exchangeType = 'direct';
        var routingKey = "celery";
        // const serializer = 'json';
        this.isReady().then(function () {
            return _this.broker.publish(body, exchange, routingKey, headers, properties);
        });
    };
```

Reply:
@actumn any news? how should we go with the next update?

Reply:
Sorry. I'm quite busy these days.

I suppose that the compile result came from `"target": "ES5"` in `tsconfig.json` which does not have arrow function grammar. So typescript compiles the arrow function in the right way. 

If you modify `tsconfig.json` to 
```
      "target": "ES2015",
      "moduleResolution": "node",
```
then you may get the result like below:
```
    sendTaskMessage(taskName, message) {
        const { headers, properties, body /*, sentEvent */ } = message;
        const exchange = "";
        // exchangeType = 'direct';
        // const serializer = 'json';
        this.isReady().then(() => this.broker.publish(body, exchange, this.conf.CELERY_QUEUE, headers, properties));
    }
```

I think it's not a good way to modify `this` as you suggest.

And why don't you add test cases for your suggestions? 
If then, we can discuss more :)

Reply:
since this lib is for server side processes, why do you compile it for es5? make es2015 as default in the next minor release please.

Reply:
Ok, I'll do it 👍

Reply:
I updated compile results to be es6 and released `v0.5.0`. So resolve this conversation.